easybuild/tools/include.py

```diff
@@ -0,0 +1,210 @@
+#!/usr/bin/env python
+# #
+# Copyright 2015-2015 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://vscentrum.be/nl/en),
+# the Hercules foundation (http://www.herculesstichting.be/in_English)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# http://github.com/hpcugent/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+# #
+"""
+Support for including additional Python modules, for easyblocks, module naming schemes and toolchains.
+
+@author: Kenneth Hoste (Ghent University)
+"""
+import glob
+import os
+import sys
+from vsc.utils.missing import nub
+from vsc.utils import fancylogger
+
+from easybuild.tools.build_log import EasyBuildError
+# these are imported just to we can reload them later
+import easybuild.tools.module_naming_scheme
+import easybuild.toolchains
+import easybuild.toolchains.compiler
+import easybuild.toolchains.fft
+import easybuild.toolchains.linalg
+import easybuild.toolchains.mpi
+# importing easyblocks namespace may fail if easybuild-easyblocks is not available
+# for now, we don't really care
+try:
+    import easybuild.easyblocks
+    import easybuild.easyblocks.generic
+except ImportError:
+    pass
+
+
+_log = fancylogger.getLogger('tools.include', fname=False)
+
+
+PKG_INIT_BODY = """
+from pkgutil import extend_path
+__path__ = extend_path(__path__, __name__)
+"""
+
+def create_pkg(path):
+    """Write package __init__.py file at specified path."""
+    init_path = os.path.join(path, '__init__.py')
+    try:
+        # note: can't use mkdir, since that required build options to be initialised
+        if not os.path.exists(path):
+            os.makedirs(path)
+
+        # put __init__.py files in place, with required pkgutil.extend_path statement
+        # note: can't use write_file, since that required build options to be initialised
+        handle = open(init_path, 'w')
+        handle.write(PKG_INIT_BODY)
+        handle.close()
+    except (IOError, OSError) as err:
```

Comment:
this is fine, 
but it's sometimes easier to have
`with open(newfile, 'w') as handle:`